tools/indexer/cmd/indexer/main.go

```diff
@@ -0,0 +1,434 @@
+		Value:   "0x821798d7b9d57dF7Ed7616ef9111A616aB19ed64",
+	})
+	optionEtherscanKey = altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "etherscan-key",
+		Usage:   "Etherscan API key",
+		EnvVars: []string{"INDEXER_ETHERSCAN_KEY"},
+	})
+	optionInfuraRPC = altsrc.NewStringFlag(&cli.StringFlag{
+		Name:     "infura-rpc",
+		Usage:    "Infura RPC URL",
+		EnvVars:  []string{"INDEXER_INFURA_RPC"},
+		Required: true,
+	})
+	optionBeaconBase = altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "beacon-base",
+		Usage:   "Beacon API base URL",
+		EnvVars: []string{"INDEXER_BEACON_BASE"},
+		Value:   "https://beaconcha.in/api/v1",
+	})
+	optionBlockInterval = altsrc.NewDurationFlag(&cli.DurationFlag{
+		Name:    "block-interval",
+		Usage:   "interval between block processing",
+		EnvVars: []string{"INDEXER_BLOCK_INTERVAL"},
+		Value:   12 * time.Second,
+	})
+
+	optionValidatorDelay = altsrc.NewDurationFlag(&cli.DurationFlag{
+		Name:    "validator-delay",
+		Usage:   "delay before fetching validator data",
+		EnvVars: []string{"INDEXER_VALIDATOR_DELAY"},
+		Value:   1500 * time.Millisecond,
+	})
+
+	optionBackfillEvery = altsrc.NewDurationFlag(&cli.DurationFlag{
+		Name:    "backfill-every",
+		Usage:   "interval for backfill operations",
+		EnvVars: []string{"INDEXER_BACKFILL_EVERY"},
+		Value:   5 * time.Minute,
+	})
+
+	optionBackfillLookback = altsrc.NewIntFlag(&cli.IntFlag{
+		Name:    "backfill-lookback",
+		Usage:   "number of slots to look back for backfill",
+		EnvVars: []string{"INDEXER_BACKFILL_LOOKBACK"},
+		Value:   512,
+	})
+
+	optionBackfillBatch = altsrc.NewIntFlag(&cli.IntFlag{
+		Name:    "backfill-batch",
+		Usage:   "batch size for backfill operations",
+		EnvVars: []string{"INDEXER_BACKFILL_BATCH"},
+		Value:   50,
+	})
+
+	optionHTTPTimeout = altsrc.NewDurationFlag(&cli.DurationFlag{
+		Name:    "http-timeout",
+		Usage:   "HTTP client timeout",
+		EnvVars: []string{"INDEXER_HTTP_TIMEOUT"},
+		Value:   15 * time.Second,
+	})
+)
+
+func createOptionsFromCLI(c *cli.Context) *config.Config {
+	return &config.Config{
+		BlockTick:        c.Duration("block-interval"),
+		ValidatorWait:    c.Duration("validator-delay"),
+		BackfillEvery:    c.Duration("backfill-every"),
+		BackfillLookback: int64(c.Int("backfill-lookback-slots")),
+		BackfillBatch:    c.Int("backfill-batch"),
+		HTTPTimeout:      c.Duration("http-timeout"),
+		OptInContract:    c.String("opt-in-contract"),
+		EtherscanKey:     c.String("etherscan-api-key"),
+		InfuraRPC:        c.String("infura-rpc"),
+		BeaconBase:       c.String("beacon-base"),
+	}
+}
+
+func startIndexer(c *cli.Context) error {
+
+	dbURL := c.String(optionDatabaseURL.Name)
+	infuraRPC := c.String(optionInfuraRPC.Name)
+	beaconBase := c.String(optionBeaconBase.Name)
+	// Initialize random seed
+	rand.Seed(time.Now().UnixNano())
+
+	// Load configuration
+
+	// Validate required configuration
+
+	log.Printf("[INIT] Starting blockchain indexer with StarRocks database")
```

Comment:
In all other parts of the code we use slog. This outputs structured json logs which are easier to parse. It has option for text logging also and it can be configured. The other services also use this and we have some util pkg also. Better to start using this in the codebase.

All this manual `[COMPONENT]` can be added using log tags. This gives us the ability to filter using the tag in the logs instead of doing a text search.